crates/abi-decoder/src/cursor.rs

```diff
@@ -39,8 +39,8 @@ impl<'a> Cursor<'a> {
         if self.is_eof() {
             Option::None
         } else {
-            let byte = unsafe { *self.offset_ptr() };
-            Option::Some(byte)
+            let byte = unsafe { self.bytes.get_unchecked(self.offset) };
```

Comment:
@neysofu IIRC, you said that using the `get_unchecked` makes the emitted Wasm code Fixed-Gas Wasm compliant. Otherwise, I don't have any other comment to add to this PR, and you can go ahead and merge it!